File: src/main/java/me/lukiiy/discordBridge/api/CommandPlate.java
package me.lukiiy.discordBridge.api;

import net.dv8tion.jda.api.interactions.commands.CommandInteraction;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

public interface CommandPlate {
    CommandData command();
    default void interaction(CommandInteraction i) {
        i.reply("Hey there!").setEphemeral(true).queue();
    }
}

File: common/src/main/kotlin/me/lukiiy/discordBridge/api/CommandPlate.kt
package me.lukiiy.discordBridge.api

import net.dv8tion.jda.api.interactions.commands.CommandInteraction
import net.dv8tion.jda.api.interactions.commands.build.CommandData

/**
 * A simple Discord command plate
 */
interface CommandPlate {
    fun command(): CommandData

    fun interaction(interaction: CommandInteraction)
}
